src/error.jl

```diff
@@ -180,7 +180,13 @@ immutable CLError <: Exception
     desc::Symbol
 
     function CLError(c::Integer)
-        @compat new(c, _cl_error_codes[Int(c)])
+        desc = ""
+        try
```

Comment:
Instead of `try ... catch` it might be better to use `get(_cl_error_codes, Int(c), :UNKOWN_ERROR_CODE)